src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php

```diff
@@ -94,11 +95,23 @@ public function getProxyFactoryCode(Definition $definition, $id, $factoryCode =
      */
     public function getProxyCode(Definition $definition)
     {
-        return preg_replace(
+        $code = $this->classGenerator->generate($this->generateProxyClass($definition));
+
+        $code = preg_replace(
             '/(\$this->initializer[0-9a-f]++) && \1->__invoke\(\$this->(valueHolder[0-9a-f]++), (.*?), \1\);/',
             '$1 && ($1->__invoke(\$$2, $3, $1) || 1) && $this->$2 = \$$2;',
-            $this->classGenerator->generate($this->generateProxyClass($definition))
+            $code
         );
+
+        if (version_compare(Version::getVersion(), '2.2', '<')) {
+            $code = preg_replace(
+                '/((?:\$(?:this|initializer|instance)->)?(?:publicProperties|initializer|valueHolder))[0-9a-f]++/',
+                '${1}'.$this->getSuffix($definition),
```

Comment:
$1 for consistency

Reply:
@nicolas-grekas no. We need the braces here, because the suffix could start with a number, which would change the placeholder when concatenating the string